src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -615,11 +679,25 @@ public PublisherClientImpl(Publisher.IPublisherClient grpcClient, PublisherSetti
             };
             return s_listTopicsPageStreamer.FetchAsync(
                 request,
-                (pageStreamRequest, cancellationToken) => GrpcClient.ListTopicsAsync(
-                    pageStreamRequest,
-                    _clientHelper.BuildCallOptions(cancellationToken, callSettings)
-                ).ResponseAsync
-            );
+                (pageStreamRequest, cancellationToken) =>
+                {
+                    // if IAsyncEnumerator.MoveNext(CancellationToken) is passed a default cancellation-token
+                    // (also CancellationToken.None), then use the CancellationToken provided in callSettings.
+                    // Otherwise use the MoveNext() cancellation-token.
+                    // It's unfortunate (for us!) that MoveNext() takes a CancellationToken, rather than a
+                    // nullable CancellationToken?
+                    CallSettings useCallSettings;
```

Comment:
Yes, that may well be a less surprising solution.

However, we'll need to consider what to do with the cancellation-token that may be present at the veneer-level. Do we merge that one _as well_? Or do the per-method/per-MoveNext cancellation-token(s) continue to override the veneer-level cancellation-token?